components/homme/src/theta-l/element_ops.F90

```diff
@@ -150,17 +145,37 @@ subroutine get_pottemp(elem,pottemp,hvcoord,nt,ntQ)
   real (kind=real_kind) :: Rstar(np,np,nlev)
   integer :: k
 
-  do k=1,nlev
-     dp(:,:,k) = ( hvcoord%hyai(k+1) - hvcoord%hyai(k) )*hvcoord%ps0 + &
-          ( hvcoord%hybi(k+1) - hvcoord%hybi(k) )*elem%state%ps_v(:,:,nt)
-  enddo
   call get_R_star(Rstar,elem%state%Q(:,:,:,1))
   
-  pottemp(:,:,:) = Rgas*elem%state%vtheta_dp(:,:,:,nt)/(Rstar(:,:,:)*dp(:,:,:))
+  pottemp(:,:,:) = Rgas*elem%state%vtheta_dp(:,:,:,nt)/(Rstar(:,:,:)*elem%state%dp3d(:,:,:,nt))
   
   end subroutine get_pottemp
   
 
+  !_____________________________________________________________________
+  subroutine get_pi(pi,dp,hvcoord)
+  !
+  implicit none
+    
+  real (kind=real_kind), intent(out)  :: pi(np,np,nlev)
+  real (kind=real_kind), intent(in)   :: dp(np,np,nlev)
+  type (hvcoord_t),     intent(in)    :: hvcoord                      ! hybrid vertical coordinate struct
+  
+  integer :: k
+  real(kind=real_kind), dimension(np,np,nlevp) :: pi_i
+
+  pi_i(:,:,1)=hvcoord%hyai(1)*hvcoord%ps0
```

Comment:
see my comment about using `hyai(nlev)` vs. `hyai(1)` to get the surface pressure.